include/yaml-cpp/node/convert.h

```diff
@@ -243,29 +243,29 @@ struct convert<std::list<T> > {
 };
 
 // std::array
-template <typename T, std::size_t S>
-struct convert<std::array<T, S> > {
-  static Node encode(const std::array<T, S>& rhs) {
+template <typename T, std::size_t N>
+struct convert<std::array<T, N>> {
+  static Node encode(const std::array<T, N>& rhs) {
     Node node(NodeType::Sequence);
-    for (typename std::array<T, S>::const_iterator it = rhs.begin();
-        it != rhs.end(); ++it)
-      node.push_back(*it);
+    for (auto element : rhs) {
+      node.push_back(element);
+    }
     return node;
   }
 
-  static bool decode(const Node& node, std::array<T, S>& rhs) {
+  static bool decode(const Node& node, std::array<T, N>& rhs) {
     if (!node.IsSequence())
```

Comment:
Forgot to say before, always put {} around your "if" clauses (we've obviously not 100% consistent in yaml-cpp, but going that way, and new code should always use this).